tests/framework/db/QueryBuilderTest.php

```diff
@@ -1178,6 +1178,18 @@ public function conditionProvider()
 
             [['in', 'id', new TraversableObject([1, 2, 3])], '[[id]] IN (:qp0, :qp1, :qp2)', [':qp0' => 1, ':qp1' => 2, ':qp2' => 3]],
 
+            //in using array objects containing null value
+            [['in', 'id', new TraversableObject([1, null])], '[[id]]=:qp0 AND [[id]] IS NULL', [':qp0' => 1]],
```

Comment:
That doesn't look right. `in` is about the value is being in the list. "in X [1,2,3]" transforms into `X IN [1,2,3]` that is equal to `X = 1 OR X = 2 OR X = 3`. Here we have `AND` and I think it should be `OR`.